cmd/minikube/cmd/config/profile_list.go

```diff
@@ -149,7 +149,7 @@ func profileStatus(p *config.Profile, api libmachine.API) string {
 
 func renderProfilesTable(ps [][]string) {
 	table := tablewriter.NewWriter(os.Stdout)
-	table.SetHeader([]string{"Profile", "VM Driver", "Runtime", "IP", "Port", "Version", "Status", "Nodes"})
+	table.SetHeader([]string{"Profile", "VM Driver", "Runtime", "IP", "Port", "Version", "Status", "Nodes", "Active"})
```

Comment:
thank you could you please post the output of "minikube profile list -o=json"
and ensure the json output also is doing the right thing ?

I think the json output should say

"active:true"
or maybe
"default:true"

Reply:
@spowelljr do u think it should be called active or default?

Reply:
I think `active` is better, `default` sounds a little ambiguous to me. I think `active` makes it inherently known that another can be changed to `active`, while `default` could make the user think that's always the `default` and there's not a way to change it.

Reply:
@medyagh Should I add the field into the `config.ClusterConfig` structure or should I just modify the `body map[string]interface{}`  in [profile_list.go](https://github.com/kubernetes/minikube/blob/4dd50b40aec3398dcf57758b083fb05b1b3cda15/cmd/minikube/cmd/config/profile_list.go#L198) ?

Reply:
Hi @vharsh, the active config is already present globally in `~/.minikube/config/config.json` so it doesn't need to be added to `config.ClusterConfig`.

Reply:
@spowelljr So I should set the structure to be like, such that `config.Profile` & `config.Config.ClusterConfig` structure doesn't need changes, i.e. I just add it to the `map[string]interface{}`, correct as the other way will make the key duplicate across all configs(and due to poor structure design it'd be harder to figure out which is the default config).

```json
{
    "valid": [ {...} ],
    "invalid": [ {...} ],
    "active": "<string-name-of-the-current-minikube-profile>"
}
```

Reply:
Hi @vharsh, that's a good question.

Let me preface my response with saying that I'm imagining this output is being consumed by another application and it's being formatting it into a nice user interface where the user can check the status of the clusters.

With that said, in my opinion having `active` in the `config.Profile` makes the most sense to me.

I understand the reasoning to have `active` at the top level as it makes it easy to know which profile is active. However, my rebuttal to `it'd be harder to figure out which is the default config` is that if someone wants to know the active profile, they can simply run `minikube profile` and get that answer.

The issue I'm seeing with this solution is that we're leaving it up to the receiver to have to inject `"active": true` into the active profile if we're considering my preface. And if we look at the output (`"valid": [ {...} ]`) we can see we're returning an array of object, so this would require them to loop through the profiles until they find the profile that matches `active` and inject it. This would be an inconvenience to the user that I would like the avoid.

If we ignore `it'd be harder to figure out which is the default config` (which can be resolved with `minikube profile`) I don't see any downsides to the method of putting `active` into `config.Profile`. However, I think implementing both methods would also be an acceptable solution as it would satisfy both cases without any downside.

Let me know what you think @vharsh

Reply:
My above comment contradicts https://github.com/kubernetes/minikube/pull/14079#discussion_r864259308, as I was asked your previous question verbally by someone else and the intent of your question got lost in the process and I responded without properly reading it, sorry for any confusion.

Reply:
@spowelljr okay, I'm just re-iterating to make sure that I've got it right, I've got to

1. add an `Active` bool field in the `config.Profile` so that both the table & the JSON format have the field coming from a struct member
2. fill it with the correct active profile everywhere i.e. when an active one gets deleted, figure out which would be the next active one and fill it in; this would likely warrant a _separate PR_ as `minikube profile` can still output a non-existent default one(i.e. `minikube`)

Let me know if I got this right.

I've also observed something regarding profiles which isn't being handled right now, is this something we should be bothering about?

```bash
➜  minikube git:(profileListFix) ✗ minikube profile list
|----------|-----------|---------|--------------|------|---------|---------|-------|
| Profile  | VM Driver | Runtime |      IP      | Port | Version | Status  | Nodes |
|----------|-----------|---------|--------------|------|---------|---------|-------|
| minikube | docker    | docker  | 192.168.58.2 | 8443 | v1.20.2 | Running |     1 |
| openebs  | docker    | docker  | 192.168.49.2 | 8443 | v1.20.2 | Running |     1 |
| third    | docker    | docker  | 192.168.67.2 | 8443 | v1.20.2 | Running |     1 |
|----------|-----------|---------|--------------|------|---------|---------|-------|

➜  minikube git:(profileListFix) ✗ minikube delete -p minikube
🔥  Deleting "minikube" in docker ...
🔥  Deleting container "minikube" ...
🔥  Removing /home/harsh/.minikube/machines/minikube ...
💀  Removed all traces of the "minikube" cluster.
➜  minikube git:(profileListFix) ✗ minikube profile
minikube
# the above profile doesn't exist ^^
➜  minikube git:(profileListFix) ✗ ./out/minikube profile list
|---------|-----------|---------|--------------|------|---------|---------|-------|--------|
| Profile | VM Driver | Runtime |      IP      | Port | Version | Status  | Nodes | Active |
|---------|-----------|---------|--------------|------|---------|---------|-------|--------|
| openebs | docker    | docker  | 192.168.49.2 | 8443 | v1.20.2 | Running |     1 |        |
| third   | docker    | docker  | 192.168.67.2 | 8443 | v1.20.2 | Running |     1 |        |
|---------|-----------|---------|--------------|------|---------|---------|-------|--------|
```

Reply:
@vharsh Yes, the code looks good for this PR other than the two small comments I left.

I will note that there is an "edge" case, but I don't think we should be supporting it. But if the user runs `minikube profile list -p minikube2` `minikube2` would be listed as active while the config itself may have `minikube` set as the active profile. Like I said, it doesn't make sense for the user to do that, but just thought I'd point it out, `minikube profile -p minikube2` experiences this same issue.

In regard to the active profile, we probably need to have a larger discussion about it. As you're likely aware, if you try to set a non-existing profile using `minikube profile` it will throw the following error:
```
$ minikube profile minikube2
❌  profile "minikube1" not found
💡  if you want to create a profile you can by this command: minikube start -p minikube1
```
And as you correctly pointed out, if you set `minikube2` on an existing profile and then delete that cluster it defaults back to `minikube`. If there is a `minikube` cluster it should go that, but if not I'd be okay with it picking an existing cluster at random, it should output a message after the delete to notify the user of this change such as `Switching default profile to "minikube3"` or something like that. The one thing is there is a second way to set the default profile, such as `minikube config set profile minikube2` which doesn't check if a matching cluster exists before setting it. So with this command it would still be possible to get in the situation where none of the clusters are the actually the active one. The two choices is we decide that it's okay to not have any cluster be the active cluster, or we modify the `minikube config set profile` command to do an active check like `minikube profile` does. Either way, this should be in a separate PR, but it's a good discussion to have.